src/iterators/dijkstra.rs

```diff
@@ -71,6 +72,12 @@ impl<'a, T> Dijkstra<'a, T>
         if graph.fetch(src).is_none() {
             return Err(GraphErr::NoSuchVertex);
         }
+        
+        for edge in graph.edges() {
+            if graph.weight(edge.1, edge.0).unwrap() < 0.0 {
```

Comment:
If the edge weight is not set this will panic by unwrapping. Can you please put a case handling an unset weight? When an edge's weight is unset it means that it is 0.

Reply:
Oh, I totally forgot that. Will fix it soon.

Reply:
I've corrected the case. Turns out, I haven't considered the same scenario in one more place. Thanks for highlighting it out.